src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeIdentifier.cs

```diff
@@ -271,12 +271,13 @@ private static void EscapeKeywords(string identifier, StringBuilder sb)
             }
         }
 
+        private static readonly char[] s_identifierSeparators = new char[] {'.', ',', '<', '>'};
         [return: NotNullIfNotNull("identifier")]
         private static string? EscapeKeywords(string? identifier)
```

Comment:
This function is suboptimal in a variety of ways.  If it's on a hot path, there are other improvements that can be made.

Reply:
Maybe @krwq can say if he think that this is a hot path, I check code and see that it is used in several places also have several others which are quite similar and can be optimized. If we receive approval that it is hot - I can prepare another PR with my suggestions for improvements.

Reply:
I'm not super familiar with serialization part of the code but quick search looks like it's only used to generate C# code which is most likely a single time cost overall which doesn't sound like a hot path to me. It's not bad idea to optimize it anyway but I'd prefer to leave it for separate PR.